### PR TITLE
feat: modify signature threshold rotate keys

### PR DESCRIPTION
**Pull request by djordon**
## Description

Closes https://github.com/stacks-network/sbtc/issues/424

## Changes

* Allow the signature threshold be changed during rotate-keys contract calls.
* Add new tests given the new behavior.

## Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: contracts/contracts/sbtc-bootstrap-signers.clar [hunk 1]
```diff
@@ -27,12 +28,20 @@
...
+    (new-aggregate-pubkey (buff 33))
+    (new-signature-threshold uint)
+  )
     (let 
         (
             (current-signer-data (contract-call? .sbtc-registry get-current-signer-data))   
-            (new-signer-principal (pubkeys-to-principal new-keys signature-threshold))
+            (new-signer-principal (pubkeys-to-principal new-keys new-signature-threshold))
         )
+        ;; Check that the signature threshold is valid
```
Comment (setzeus) [inline contracts/contracts/sbtc-bootstrap-signers.clar:41]: Clarity division rounds down so just making sure that it's clear here that u1 / u2 == 0. I don't believe that it materially changes the logic here but just a NIT.

**Review by setzeus — COMMENTED** [review]
Everything LGTM except for one NIT that we should be aware of but doesn't change anything.

**Review by setzeus — APPROVED** [review]

File: contracts/contracts/sbtc-bootstrap-signers.clar [hunk 2]
```diff
@@ -27,12 +28,20 @@
...
+        (asserts! (and (> new-signature-threshold (/ (len new-keys) u2))
```
Comment (hstove) [inline contracts/contracts/sbtc-bootstrap-signers.clar:42]: Where does this "rule" come from? If we're going to let the signers determine a new threshold (in this code, between 50%-100%), why not lower?

**Review by hstove — COMMENTED** [review]
Overall the code seems valid and makes sense, but I'm uncertain how I feel regarding the comment I made about the valid threshold range.

File: contracts/contracts/sbtc-bootstrap-signers.clar (hunk 2 above)
Comment (djordon) [reply contracts/contracts/sbtc-bootstrap-signers.clar:42]: Good questions. I think I originally had it at any positive threshold and thought "we'll probably want to ensure that the threshold will never be less than 50%". So, this comes out of thin air. Either greater than 50% or greater than 0 sound defensible to me and I do not have any strong feels on either.

**Review by hstove — APPROVED** [review]

File: contracts/contracts/sbtc-bootstrap-signers.clar (hunk 2 above)
Comment (hstove) [reply contracts/contracts/sbtc-bootstrap-signers.clar:42]: Yep, I think that's fair!

**Pull request merged**